topic-operator/src/main/java/io/strimzi/operator/topic/BatchingTopicController.java

```diff
@@ -1213,16 +1219,22 @@ private void updateStatus(ReconcilableTopic reconcilableTopic) {
         
         // the observedGeneration is a marker that shows that the operator works and that it saw the last update to the resource
         reconcilableTopic.kt().getStatus().setObservedGeneration(reconcilableTopic.kt().getMetadata().getGeneration());
-        
-        // set or reset the topicName
+
+        // add topicName
         reconcilableTopic.kt().getStatus().setTopicName(
             !TopicOperatorUtil.isManaged(reconcilableTopic.kt())
                 ? null
                 : oldStatus != null && oldStatus.getTopicName() != null
-                ? oldStatus.getTopicName()
-                : TopicOperatorUtil.topicName(reconcilableTopic.kt())
+                    ? oldStatus.getTopicName()
+                    : TopicOperatorUtil.topicName(reconcilableTopic.kt())
         );
-        
+
+        // add or remove topicId
+        reconcilableTopic.kt().getStatus().setTopicId(
+            (!TopicOperatorUtil.isManaged(reconcilableTopic.kt()) || TopicOperatorUtil.isPaused(reconcilableTopic.kt()))
+                ? null : topicNameIdMap.get(reconcilableTopic.topicName())
+        );
+
         if (statusChanged(reconcilableTopic.kt(), oldStatus)) {
```

Comment:
This is no longer just checking whether the statuses differ. The logic also depends on the metadata. So let's rename it to something like `statusUpdateRequired`.